packages/headless/src/modules/order/services/customer.service.ts

```diff
@@ -32,6 +37,40 @@ export class OrderCustomerService {
     );
   }
 
+  async reOrder(userId: string, body: ReOrderQuery): Promise<any>{
+    let { ignoreInvalidItems, overWriteCart, orderId } = body; 
+
+    const order = await this.orderRepository.getOrderById(orderId);
+    const products = order.products;
+    
+    const unavailableProducts = await this.orderRepository.getAvailableProducts(products);
+    if(unavailableProducts.length === products.length)return errorResponse( 'Products are not available right now',null,HttpStatus.BAD_REQUEST );
+    else {
+        let availableAddToCart;
+        const allAddToCart = products.map(product => { return {productId: product.productId, quantity: product.quantity}});
+
+        if(unavailableProducts.length > 0){
+            if(ignoreInvalidItems === false) return errorResponse( 'All of the products are not available. Do you wish to continue?', null, HttpStatus.BAD_REQUEST );
+            
+            const res = products.filter(product => !unavailableProducts.find(item => item.productId === product.productId ));
+            availableAddToCart = res.map(product => { return { productId: product.productId, quantity: product.quantity }});
```

Comment:
you are returning the unavailable product id and quantity. how come the front-end team know those items name? because front-end will need to show the unavailable products name, instead of their ids. in this case you should return product id as well as product name. quantity is not needed to return